src/greentest/test__pool.py

```diff
@@ -204,6 +205,46 @@ def test_add_method(self):
         finally:
             first.kill()
 
+    def test_add_method_non_blocking(self):
+        p = self.klass(size=1)
+        first = gevent.spawn(gevent.sleep, 1000)
+        try:
+            second = gevent.spawn(gevent.sleep, 1000)
+            try:
+                p.add(first)
+                try:
```

Comment:
This (and possibly the next test) can be written more clearly using `with self.assertRaises(...):` I think.

Reply:
Perfect. I never use `unittest` (in favor of `pytest`), wasn't aware that it supported `assertRaises`. Thanks!